components/gcp/dataflow/launch_flex_template/README.md

```diff
@@ -0,0 +1,159 @@
+to run a Dataflow pipeline that reads the [Shakespeare word index](https://cloud.google.com/bigquery/public-data#sample_tables)
+sample public dataset and writes the data in Parquet format to the user provided GCS bucket.
+
+## Caution & requirements
+To use the component, the following requirements must be met:
+- Cloud Dataflow API is enabled.
+- The component can authenticate to GCP. Refer to [Authenticating Pipelines to GCP](https://www.kubeflow.org/docs/gke/authentication-pipelines/) for details.
+- The output cloud storage bucket must exist before running the pipeline.
+- The Kubeflow user service account is a member of:
+    - `roles/dataflow.developer` role of the project.
+    - `roles/storage.objectCreator` role of the Cloud Storage Object for `staging_dir.` and output data folder.
+- The dataflow controller service account is a member of:
+    - `roles/bigquery.readSessionUser` role to create read sessions in the project.
+    - `roles/bigquery.jobUser` role to run jobs including queries.
+    - `roles/bigquery.dataViewer` role to read data and metadata from the table of view
+
+
+---
+
+### Follow these steps to use the component in a pipeline:
+
+#### 1. Install the Kubeflow Pipeline SDK:
+
+```python
+%%capture --no-stderr
+
+!pip3 install kfp --upgrade
+```
+
+#### 2. Load the component using KFP SDK
+```python
+import kfp.components as comp
+
+dataflow_template_op = comp.load_component_from_url(
+    'https://raw.githubusercontent.com/kubeflow/pipelines/1.5.0-rc.0/components/gcp/dataflow/launch_flex_template/component.yaml')
+help(dataflow_template_op)
+```
+
+#### 3. Configure job parameters
+```python
+PROJECT_ID = '[Your PROJECT_ID]'
+BIGQUERY_TABLE_SPEC = '[Your PROJECT_ID:DATASET_ID.TABLE_ID]'
+GCS_OUTPUT_FOLDER = 'gs://[Your output GCS folder]'
+GCS_STAGING_FOLDER = 'gs://[Your staging GCS folder]'
+LOCATION = 'us'
+# Optional Parameters
+EXPERIMENT_NAME = 'Dataflow - Launch Flex Template'
+
+flex_temp_launch_parameters = {
+    "parameters": {
+        "tableRef": BIGQUERY_TABLE_SPEC,
+        "bucket": GCS_OUTPUT_FOLDER
+    },
+    "containerSpecGcsPath": "gs://dataflow-templates/2021-03-29-00_RC00/flex/BigQuery_to_Parquet",
+}
+```
+
+#### 4. Example pipeline that uses the component
+
+```python
+import kfp.dsl as dsl
+import json
+@dsl.pipeline(
+    name='Dataflow launch flex template pipeline',
+    description='Dataflow launch flex template pipeline'
+)
+def pipeline(
+    project_id = PROJECT_ID,
+    location = LOCATION,
+    launch_parameters = json.dumps(flex_temp_launch_parameters),
+    validate_only = 'False',
+    staging_dir = GCS_STAGING_FOLDER,
+    wait_interval = 30):
+    dataflow_template_op(
+        project_id = project_id,
+        location = location,
+        launch_parameters = launch_parameters,
+        validate_only = validate_only,
+        staging_dir = staging_dir,
+        wait_interval = wait_interval)
+
+```
+
+#### 5. Compile the pipeline
```

Comment:
Could we use `client.create_run_from_pipeline_func` instead of explicitly compiling and submitting?